a10_octavia/controller/worker/tasks/vthunder_tasks.py

```diff
@@ -200,7 +200,7 @@ def execute(self, added_ports, loadbalancer, vthunder):
             compute_id = loadbalancer.amphorae[0].compute_id
             network_driver = utils.get_network_driver()
             nics = network_driver.get_plugged_networks(compute_id)
-            if len(added_ports[amphora_id]) > 0:
+            if added_ports and amphora_id in added_ports and len(added_ports[amphora_id]) > 0:
```

Comment:
Why the check amphora_id in added_ports is added here?
I think this check is not required here because if added_ports is there then it must be having the amphora_id in it as for each amp_id we define added_ports.
https://github.com/a10networks/a10-octavia/blob/master/a10_octavia/controller/worker/tasks/a10_network_tasks.py#L294